optaplanner-core/src/main/java/org/optaplanner/core/api/stat/ScoreCalculationSpeedSubSingleStatisticListener.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.api.stat;
+
+import org.optaplanner.core.impl.phase.event.PhaseLifecycleListenerAdapter;
+import org.optaplanner.core.impl.phase.scope.AbstractPhaseScope;
+import org.optaplanner.core.impl.phase.scope.AbstractStepScope;
+import org.optaplanner.core.impl.solver.scope.SolverScope;
+
+public class ScoreCalculationSpeedSubSingleStatisticListener<Solution_> extends PhaseLifecycleListenerAdapter<Solution_> {
+
+    private final long timeMillisThresholdInterval = 10;
+
+    private long nextTimeMillisThreshold = timeMillisThresholdInterval;
+    private long lastTimeMillisSpent = 0L;
+    private long lastCalculationCount = 0L;
+
+    private long scoreCalculationCount = 0L;
+    private long scoreCalculationSpeed = 0L;
+
+    public long getScoreCalculationSpeed() {
+        return scoreCalculationSpeed;
+    }
+
+    public long getScoreCalculationCount() {
+        return scoreCalculationCount;
+    }
+
+    @Override
+    public void stepEnded(AbstractStepScope<Solution_> stepScope) {
+        long timeMillisSpent = stepScope.getPhaseScope().calculateSolverTimeMillisSpentUpToNow();
+        if (timeMillisSpent >= nextTimeMillisThreshold) {
+            SolverScope<Solution_> solverScope = stepScope.getPhaseScope().getSolverScope();
+            scoreCalculationCount = solverScope.getScoreCalculationCount();
+            long calculationCountInterval = scoreCalculationCount - lastCalculationCount;
+            long timeMillisSpentInterval = timeMillisSpent - lastTimeMillisSpent;
```

Comment:
Can't micrometer do this for us? To simply take the metric and average it over time?
I would expect this to be the case.

Reply:
Given https://micrometer.io/docs/concepts#rate-aggregation and https://micrometer.io/docs/concepts#_counters probably, but it seems to be an operation that needed to be done on the measurement consumer.

Reply:
For the moment, it will be easier to just keep it as is. (Especially if we want to consume these metrics from benchmarker.) But over time, I'd rather see us use micrometer as opposed to doing things like timing ourselves.